src/components/qas/questions/qaCalculators/sizeCalculator/index.jsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react/no-danger */
+import React from 'react';
+import { CardText, CardActions } from 'react-md';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import classnames from 'classnames';
+// import Calculation from '../shared/calculation';
+import CalculatedMeasurement from '../shared/CalculatedMeasurement';
+import Equation from '../shared/equation';
+import TextField from '../../../../site/forms/textField';
+import Card from '../../../../site/card';
+import { renderDef, formatValue } from '../../../../../lib/utilities.js';
+import './sizeCalculator.module.scss';
+
+const HIcon = () => {
+  return (
+    <div>
+      <strong>H</strong> =
+    </div>
+  );
+};
+
+const PIcon = () => {
+  return (
+    <div>
+      <strong>p</strong> =
+    </div>
+  );
+};
+
+class SizeCalculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: undefined,
+      cardActive: false,
+      hasFocus: false,
+      answerable: false,
+    };
+  }
+
+  componentDidMount() {
+    const { answerable, value } = this.state;
+    const { question, activeId, answer } = this.props;
+    const { id } = question;
+    const calculateValue = isEmpty(answer) ? value : answer.content;
+
+    this.checkAnswerable(answerable, activeId === id);
+    this.updateCalculatedMeasurements(calculateValue);
+  }
+
+  componentDidUpdate() {
+    const { answerable } = this.state;
+    const { question, activeId } = this.props;
+    const { id } = question;
+
+    this.checkAnswerable(answerable, activeId === id);
+  }
+
+  checkAnswerable(answerable, active) {
+    if (!answerable && active) {
+      this.setState(prevState => ({
+        ...prevState,
+        answerable: true,
+      }));
+    }
+  }
+
+  calculateDiameter(value) {
+    if (value && value.h && value.p) {
+      const diameter = formatValue(
+        (1329 / Math.sqrt(value.p)) * 10 ** (-0.2 * value.h),
+        3
+      );
+      return diameter;
+    }
+    return null;
+  }
+
+  updateCalculatedMeasurements(value) {
+    const diameter = this.calculateDiameter(value);
+    this.setState(prevState => ({
+      ...prevState,
+      value: { ...value, d: diameter },
+    }));
+  }
+
+  handleHChange = value => {
+    this.handleChange(value, 'h');
+  };
+
+  handlePChange = value => {
+    this.handleChange(value, 'p');
+  };
+
+  handleChange = (value, index) => {
+    const { question, answerHandler } = this.props;
+    const { id } = question;
+    const { value: oldValue } = this.state;
+    const newVal = {
+      ...oldValue,
+      [index]: value,
+    };
+
+    if (newVal.h && newVal.p && newVal.d) {
```

Comment:
I am fairly certain this will never be true when you want it to be.  You're checking h, p, and d on the newVal.  I believe this will always yield the newest h and p, but an old d, such that even if the condition is met it will pass through inaccurate values.  

I think probably what you want to do is either `updateCalculatedMeasurements` and then, peak at updated state, and then pass `answerHandler` the updated `value` on the updated state, 
OR
 `calculateDiameter` here, include it in `newVal`, pass it on to `answerHandler`, and update `updateCalculatedMeasurements` to NOT calculate the diameter, but instead just be in charge of updating your state.